pkg/logger/zerologr.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Jorn Friedrich Dreyer
+// Modified 2021 Serhii Mikhno
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package zerologr defines an implementation of the github.com/go-logr/logr
+// interfaces built on top of zerolog (github.com/rs/zerolog).
+
+package logger
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"github.com/rs/zerolog"
+)
+
+const (
+	debugVerbosity = 2
+	traceVerbosity = 8
+	timeFormat     = "2006-01-02 15:04:05.000"
+)
+
+var (
+	log Zerologr
+	mu  sync.RWMutex
+)
+
+// this prevents from using non-initialised logger
+func init() {
```

Comment:
Let's avoid global side-effects.  This calls Init which calls zerolog which will result in a bad time for any client code that imports this module and has already done their own zerolog configuration.